src/main/java/me/champeau/gradle/japicmp/JApiCmpWorkerAction.java

```diff
@@ -197,9 +193,14 @@ private void generateOutput(JarArchiveComparator jarArchiveComparator) {
     Options options = Options.newDefault();
     options.setOldClassPath(japicmp.util.Optional.of(toClasspath(oldClasspath)));
     options.setNewClassPath(japicmp.util.Optional.of(toClasspath(newClasspath)));
-    final List<JApiCmpArchive> baseline = toJapiCmpArchives(oldArchives);
-    final List<JApiCmpArchive> current = toJapiCmpArchives(newArchives);
-    List<JApiClass> jApiClasses = jarArchiveComparator.compare(baseline, current);
+    Diff diff = new Diff(jarArchiveComparator, oldArchives, newArchives);
+    List<JApiClass> jApiClasses = diff.classes();
+
+    if (compatibilityChangesFilterFile != null) {
+      CompatibilityChangesFilter filter = new CompatibilityChangesFilter(compatibilityChangesFilterFile);
+      filter.filterChanges(diff);
+    }
```

Comment:
Do I get it right that this piece of code actually modifies instances of `JApiClass`, so `List<JApiClass> jApiClasses = diff.classes();` is affected?

Reply:
Yeah, this code modify the instances of JApiClass but the LIST is not modified.